src/quest/persistence.py

```diff
@@ -35,15 +38,20 @@ def __init__(self, namespace: str, storage: BlobStorage):
         if storage.has_blob(namespace):
             self._keys = storage.read_blob(namespace)
             for key in self._keys:
-                self._items.append(storage.read_blob(key))
+                serialized_item = self._storage.read_blob(key)
+                # Deserialize the item that is read from storage
+                item = asyncio.run(self._master_serializer.deserialize(serialized_item))
```

Comment:
`asyncio.run` can only be run once in the application (and is typically always at the root). Let's discuss this in class.